server/services/conditionService.ts

```diff
@@ -254,4 +257,19 @@ export default class ConditionService {
     return mappedConditions
   }
   /* eslint-disable no-param-reassign */
+
+  async getParentOrSelfAdditionalLicenceConditions(licence: Licence, user: User): Promise<AdditionalCondition[]> {
```

Comment:
Yes @SebNegahbanMoJ , specifically suggested to pull this down (not sure about the name though!)

Controllers should be very light, ideally just convert to and from request/response and then delegate down to the service layer. 
 
There was far too much logic in this situation.

Reply:
> mainly because it's not really reflective of what's happening inside

We use abstraction to avoid exposing "the how".

Ideally we'd express this in a more domain specific way: What are we showing here?

 "getRelevantConditions",  "getConditionsToCheck"... probably too abstract...

Reply:
(not a big fan of `getParentLicenceOrSelf` for the same reason)

Reply:
That makes sense, I was very much pushing logic into the controller but if it's better practice to just have a call to the service layer that says 'Get me what I want', that's completely fine.
If we want to go with the hyper-specific service methods, I'd probably advocate for something like
`getAdditionalConditionsForSummaryAndPdf`? That way it's specific in its use but without getting into the detail of what is happening within the method

Reply:
This is also only getting AP conditions right? The PSS are contained within `licence.additionalPssConditions`? If that's the case, I'd also advocate for specifying `AP` in the method name, something like `getAdditionalAPConditionsForSummaryAndPdf`

Reply:
Are you certain that it's fetching the PSS conditions? The `Licence` object type appears to have:
``` 
additionalLicenceConditions: components['schemas']['AdditionalCondition'][]
additionalPssConditions: components['schemas']['AdditionalCondition'][]
```
and the nunjucks component that renders the PSS conditions appears to read them directly from the licence (which I believe is defined in the njk files by middleware) rather than from the additionalConditions object that's passed in to the checkAnswers view (PSS component is [here](https://github.com/ministryofjustice/create-and-vary-a-licence/blob/27d3529fc716a3fb015003320d600ef0da6a6fbd/server/views/partials/additionalPssConditionsSummary.njk#L49))